### DISCUSSION TITLE
Support videos in article viewer

### DISCUSSION THREAD

**Discussion by wvffle** [jocmp/capyreader#210] [Feature Requests]
The app seems promising.
The only two things that I may consider missing are #199 and ability to play videos embedded in the article.

Currently they're displayed as a play icon but after clicking it does nothing

Tested on Factorio blog: https://www.factorio.com/blog/rss

![Screenshot_20240727-113625_Capy Reader](https://github.com/user-attachments/assets/44c3397f-1b35-4dcf-a4ef-4e131d1d5faa)

**Comment by jocmp** [comment]
thanks for raising, I'll track something for videos in #199.

**Comment by jocmp** [comment]
Basic full screen video support is available as of [2024.08.1021-dev](https://github.com/jocmp/capyreader/releases/tag/2024.08.1021-dev).

**Reply by wvffle** [reply]
@jocmp I'm testing this build now with yesterday's post in the linked RSS feed.

In the default display mode, only first video is recognized as the video and can be played by clicking on the play button, although fullscreen seems to be disabled. 

The rest of the videos are not recognized and only the play icon is available as a link to the video resource. These buttons seem to have different sizes.

In the full content mode none of the videos are recognized.


<details>

![image](https://github.com/user-attachments/assets/8b2ff09f-13c7-42ed-aa77-d9e3b475ad05)
![image](https://github.com/user-attachments/assets/a061cbaf-697f-4415-a192-e3873c718e17)

</details>

**Reply by jocmp** [reply]
Thanks for the details. I'll see what I can do.

**Reply by jocmp** [reply]
I played around with full screen support. It's pretty janky and I'm not happy with the solution. For instance, the scroll and video state reset when you rotate the device.

So I went back to the drawing board and compared [NetNewsWire](https://github.com/Ranchero-Software/NetNewsWire) and [Reeder](https://reederapp.com/)'s video players. Both readers only allow inline video.

**NetNewsWire**
![Screenshot 2024-08-03 at 3 53 03 PM](https://github.com/user-attachments/assets/cbee0c57-19d3-43aa-a090-ef61e85ee6b7)

**Reeder**
![Screenshot 2024-08-03 at 3 53 16 PM](https://github.com/user-attachments/assets/a82fe3a2-7010-4036-acbd-4aa8291dd71b)

I was missing a few settings to play videos inline. I've added those to https://github.com/jocmp/capyreader/pull/229. Long story short, I'll support inline video and skip full screen videos to avoid that technical complexity.

The final result shown below. Videos won't autoplay in Full Content but they'll have all the controls to play and pause.

https://github.com/user-attachments/assets/291b34d9-1382-437d-9a47-56ecec4030bc

**Reply by jocmp** [reply]
Changes are live in [the latest release](https://github.com/jocmp/capyreader/releases/tag/2024.08.1022-dev). I also made sure that looped autoplay videos work for factorio.com. Here's what that looks like 

https://github.com/user-attachments/assets/576ec7da-5986-4799-b492-d01f277b0dc5

**Reply by wvffle** [reply]
I cannot replicate the autoplay shown in the last video with the latest github release but I am now able to play those videos inline just fine

**Comment by wvffle** [comment]
Some QOL features to consider:
- Show thumbnail of the video
- Auto pause the video if it's out of the view (especially if the video has audio track)

**Comment by PhilC813** [comment]
Twitter posts embedded in articles are rendered in full content mode by Capy, but the videos included in some of these posts don't make it. One example [here](https://www.androidauthority.com/android-xr-samsung-project-moohan-live-demo-3508533/).

Do you think it would be possible to have them show up? Although I fear it may require JavaScript.

**Reply by jocmp** [reply]
It might be possible - sometimes the JavaScript will work if it's embedded in an iframe. This is how YouTube videos work in the app for example.

I have an outstanding ticket to investigate embed. I added Twitter to the list

- https://github.com/jocmp/capyreader/issues/676
